src/analysis/retail/shaman/enhancement/modules/normalizers/MaelstromWeaponCastNormalizer.tsx

```diff
@@ -16,27 +16,26 @@ class MaelstromWeaponCastNormalizer extends EventsNormalizer {
     super(options);
 
     // this normalizer depends on the event link normalizer to have already run, so setting a lower priority to enforce later execution (higher value = lower priority)
-    this.priority = 10;
+    this.priority = NormalizerOrder.MaelstromWeaponCastNormalizer;
   }
 
   normalize(events: AnyEvent[]): AnyEvent[] {
     const fixedEvents: AnyEvent[] = [];
-    events.forEach((event: AnyEvent) => {
+    events.forEach((event: AnyEvent, index: number) => {
       if (HasAbility(event) && MAELSTROM_WEAPON_ELIGIBLE_SPELL_IDs.includes(event.ability.guid)) {
         if (
           event.type === EventType.BeginCast ||
           event.type === EventType.BeginChannel ||
           event.type === EventType.EndChannel
         ) {
-          if (!HasRelatedEvent(event, MAELSTROM_WEAPON_INSTANT_CAST)) {
-            fixedEvents.push(event);
+          // check the next event, if it's a cast success and the same spell as this one, skip this event
+          const next = events.at(index + 1);
+          if (next?.type === EventType.Cast && next.ability.guid === event.ability.guid) {
+            return;
```

Comment:
just want to throw in my 2c that it is worth being cautious about dropping events out of the event list. i am not sure how complex other options are so i'm not gonna be like "you must do this" but omitting multiple different event types (this + maelstrom duplicated refreshes) starts getting kinda spooky imo

Reply:
It makes later normalizers significantly easier, as I can reliably assume that a refresh is a waste. The ResourcesNormalizer that runs later on removes **_all_** maelstrom weapon buff events anyway as they're all replaced with resourcechange or as spell costs